File: JAVA/Pila/PrincipalPila.java
package JAVA.Pila;

import java.util.InputMismatchException;
import java.util.Scanner;

public class PrincipalPila {
    public static void main(String[] args) {
        Pila pila = new Pila();
        Scanner key = new Scanner(System.in);
        
        int op = 0;
        do {
            System.out.println("******MENU CARRO*******");
            System.out.println("1. Agregar");
            System.out.println("2. Buscar");
            System.out.println("3. Ordenar");
            System.out.println("4. Desordenar");
            System.out.println("5. Eliminar");
            System.out.println("6. Insertar");
            System.out.println("7. Recorrer coleccion de INICIO A FIN");
            System.out.println("8. Recorrer coleccion de FIN A INICIO");
            System.out.println("9. Imprimir colección");
            System.out.println("10. Salir");
            System.out.println("******MENU CARRO*******");

            try{
                System.out.print(">");
                op = key.nextInt();
            }catch(InputMismatchException e){
                System.out.println("Opcion invalida");
                key = new Scanner(System.in);
            }
            
            switch (op) {
                case 1:
                    System.out.println("**********AGREGAR**********");
                    System.out.print("Ingrese el marca del carro: ");
                    String marca = key.next();
                    System.out.print("Ingrese el modelo del carro: ");
                    String modelo = key.next();
                    System.out.print("Ingrese el color del carro: ");
                    String color = key.next();
                    System.out.print("Ingrese la placa del carro: ");
                    String placa = key.next();
                    Carro nuevo = new Carro(marca, modelo, color, placa.toUpperCase());
                    pila.agregar(nuevo);
                    System.out.println("**********AGREGAR**********");
                    break;

                case 2:
                    System.out.println("**********BUSCAR**********");
                    System.out.print("Ingrese la placa del carro: ");
                    String busqueda = key.next();
                    int indice = pila.buscarPorPlaca(busqueda);
                    if (indice != -1){
                        System.out.println("El carro se encuentra en la posición ["+indice+"]");
                        pila.imprimir(indice);
                    }else{
                        System.out.println("El carro no se encuentra en la colección");
                    }
                    System.out.println("**********BUSCAR**********");
                    break;

                case 3:
                    System.out.println("**********ORDENAR**********");
                    pila.ordenar();
                    System.out.println("PILA ORDENADA POR PLACA");
                    System.out.println("**********ORDENAR**********");
                    break;
                
                case 4:
                    System.out.println("**********DESORDENAR**********");
                    pila.desordenar();
                    System.out.println("PILA DESORDENADA POR PLACA");
                    System.out.println("**********DESORDENAR**********");
                    break;
                
                case 5:
                    System.out.println("**********ELIMINAR**********");
                    System.out.print("Ingrese la placa del carro: ");
                    String eliminar = key.next();
                    pila.eliminar(eliminar.toUpperCase());
                    System.out.println("**********ELIMINAR**********");
                    break;

                case 6:
                    System.out.println("**********INSERTAR**********");
                    System.out.print("Ingrese el marca del carro: ");
                    String marcaIn = key.next();
                    System.out.print("Ingrese el modelo del carro: ");
                    String modeloIn = key.next();
                    System.out.print("Ingrese el color del carro: ");
                    String colorIn = key.next();
                    System.out.print("Ingrese la placa del carro: ");
                    String placaIn = key.next();
                    Carro nuevoIn = new Carro(marcaIn, modeloIn, colorIn, placaIn.toUpperCase());
                    System.out.print("Ingrese la posición donde desea insertar el Carro: ");
                    int pos = key.nextInt();
                    pila.insertar(pos, nuevoIn);
                    System.out.println("**********INSERTAR**********");
                    break;

                case 7:
                    System.out.println("**********RECORRER DE INICIO A FIN**********");
                    pila.recorrerInicioFin();
                    System.out.println("**********RECORRER DE INICIO A FIN**********");
                    break;

                case 8:
                    System.out.println("**********RECORRER DE FIN A INICIO**********");
                    pila.recorrerFinInicio();
                    System.out.println("**********RECORRER DE FIN A INICIO**********");
                    break;

                case 9:
                    System.out.println("**********IMPRIMIR**********");
                    pila.imprimir();
                    System.out.println("**********IMPRIMIR**********");
                    break;
                default:
                    break;
            }
        } while (op != 10);
    }
}

File: PYTHON/Pila/PrincipalPila.py
from Pila import Pila
from Carro import Carro

pila = Pila()

ctrl = True
while ctrl:
    opc = 0
    print("******MENU CARRO*******")
    print("1. Agregar")
    print("2. Buscar")
    print("3. Ordenar")
    print("4. Desordenar")
    print("5. Eliminar")
    print("6. Insertar")
    print("7. Recorrer IMPRIMIR de INICIO A FIN")
    print("8. Recorrer IMPRIMIR de FIN A INICIO")
    print("9. Imprimir colección")
    print("10. Salir")
    print("******MENU CARRO*******")
    
    try:
        opc = int(input("Ingrese una opcion: "))
    except ValueError:
        print("******ERROR*******")
        print("Opcion invalida")
        print("******ERROR*******")
        input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")

    if opc == 1:
        print("**********AGREGAR**********")
        marca = input("Ingrese el marca del carro: ")
        modelo = input("Ingrese el modelo del carro: ")
        color = input("Ingrese el color del carro: ")
        placa = input("Ingrese la placa del carro: ")
        
        nuevo = Carro(marca, modelo, color, placa.upper())
        pila.agregar(nuevo)
        print("CARRO AGREGADO A LA PILA")
        input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")
        print("**********AGREGAR**********")

    elif opc == 2:
        print("**********BUSCAR**********")
        placa = input("Ingrese la placa del carro a buscar: ")
        busqueda = pila.buscar(placa)

        if busqueda == None:
            print("NO HA INGRESADO NINGUNA PLACA")
        
        else:
            print("El Carro de placa "+busqueda.getPlaca()+" "+busqueda.getMarca()+" "+busqueda.getModelo()+" "+busqueda.getColor())
            input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")
            print("**********BUSCAR**********")
        
    elif opc == 3:
        print("**********ORDENAR**********")
        pila.ordenar()
        print("PILA ORDENADA POR PLACA DE MENOR A MAYOR")
        input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")
        print("**********ORDENAR**********")

    elif opc == 4:
        print("**********DESORDENAR**********")
        pila.desordenar()
        print("PILA DESORDENADA")
        input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")
        print("**********DESORDENAR**********")

    elif opc == 5:
        print("**********ELIMINAR**********")
        placa = input("Ingrese la placa del carro a eliminar: ")
        if(placa == None):
            print("NO HA INGRESADO NINGUNA PLACA")
        else:
            eliminado = pila.eliminar(placa)
            if eliminado == -1:
                print("NO EXISTE EL CARRO CON ESA PLACA")
            else:
                print("EL CARRO DE PLACA "+placa+" HA SIDO ELIMINADO")
                input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")
                print("**********ELIMINAR**********")

    elif opc == 6:
        print("**********INSERTAR**********")
        marca = input("Ingrese la marca del carro: ")
        modelo = input("Ingrese el modelo del carro: ")
        color = input("Ingrese el color del carro: ")
        placa = input("Ingrese la placa del carro: ")
        pos = int(input("Ingrese la posición donde desea insertar el carro: "))

        nuevo = Carro(marca, modelo, color, placa.upper())
        
        if(pos >= 0 and pos <= pila.size):
            pila.insertar(pos, nuevo)
            print("CARRO INSERTADO EN LA POSICION ",pos)
            input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")
            print("**********INSERTAR**********")
        else:
            print("******ERROR*******")
            print("Posición invalida")
            print("******ERROR*******")
            input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")
            print("**********INSERTAR**********")
    
    elif opc == 7:
        print("**********RECORRER DE INICIO A FIN**********")
        pila.recorrerInicioFin()
        input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")
        print("**********RECORRER DE INICIO A FIN**********")

    elif opc == 8:
        print("**********RECORRER DE FIN A INICIO**********")
        pila.recorrerFinInicio()
        input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")
        print("**********RECORRER DE FIN A INICIO**********")

    elif opc == 9:
        print("**********IMPRIMIR**********")
        pila.imprimir()
        print("**********IMPRIMIR**********")
        input("PRESIONE UNA TECLA PARA ENTER/INTRO PARA IR AL MENU...")
        print("**********IMPRIMIR**********")

    elif opc == 10:
        ctrl = False